### ISSUE TITLE
(1 .. -Inf).reverse is wrong

### ISSUE THREAD

**Issue by AlexDaniel** [rakudo/rakudo#3595]
Code:
```raku
say (1 .. -Inf).reverse[^10]
```
Result:
```
(Inf Inf Inf Inf Inf Inf Inf Inf Inf Inf)
```

The right answer is ```(Nil Nil Nil Nil Nil Nil Nil Nil Nil Nil)``` because the endpoint is smaller (it's negative infinity, and `..` doesn't count down).

Output on all releases: https://gist.github.com/aa6da581c5d910b8ee2016c9397e3dea

Bisected: (2016-03-16) https://github.com/rakudo/rakudo/commit/ec7d8d4cc2358a4108bdab2c11ba1b57924bfc39

**Comment by lizmat** [comment]
Fixed with 3983b4f8f2

**Comment by lizmat** [comment]
Test added with https://github.com/Raku/roast/commit/3a463fe779
